userspace/falco/application.h

```diff
@@ -36,15 +43,64 @@ namespace app {
 
 class application {
 public:
+	class action_state {
+	public:
+		action_state();
+		virtual ~action_state();
+
+		bool restart;
+		bool terminate;
+		bool reopen_outputs;
+
+		std::shared_ptr<falco_configuration> config;
+		std::shared_ptr<falco_outputs> outputs;
+		std::shared_ptr<falco_engine> engine;
+		std::shared_ptr<sinsp> inspector;
+		std::set<std::string> enabled_sources;
+
+		// The event source is syscall by default. If an input
+		// plugin was found, the source is the source of that
+		// plugin.
+		std::string event_source;
```

Comment:
With the changes in https://github.com/falcosecurity/falco/pull/1944, this has been restructured a bit, but the state of falco_init on the master branch seems to use enabled_sources only to find the case where all event sources were disabled via repeated calls to `--disable_source`.

I think a good time to fix this will be when we expand falco to handle multiple plugin event sources at once, or syscall + plugin event sources side-by-side.